src/layout/layout.rs

```diff
@@ -1212,5 +1277,288 @@ mod tests {
                 ]
             );
         }
+
+        #[test]
+        fn proportional_fixed() {
+            // fixed doesn't panic when results don't match exactly
+            let [a, b, c] = Rect::new(0, 0, 100, 1).split(&Layout::horizontal([
+                Fixed(33),
+                Fixed(33),
+                Fixed(33),
+            ]));
+            assert_eq!([a.width, b.width, c.width], [33, 33, 34]);
+
+            // cassowary implementation tends to put excess in last variable
+            let [a, b, c] = Rect::new(0, 0, 100, 1).split(&Layout::horizontal([
+                Fixed(25),
+                Fixed(25),
+                Fixed(25),
+            ]));
+            assert_eq!([a.width, b.width, c.width], [25, 25, 50]);
```

Comment:
This still seems a bit confusing. I know we have segment_size, but it really feels weird that Fixed doesn't have higher priority than that.

Reply:
What are you intending this should do? Something like this?

```rust
let [a, b, c] = Rect::new(0, 0, 100, 1).split(&Layout::horizontal([
    Fixed(25),
    Fixed(25),
    Fixed(25),
]));
assert_eq!([a.width, b.width, c.width], [33, 34, 33]);
```

I have a solution for this (by introducing a invisible `Rect` at the end) but it'll have to be a separate PR. That will  change the behavior of all the tests. We would also define an alignment on the layout itself. Something like `Layout::vertical(constraints).aligment(Alignment::Start)` where alignment could be `Start`, `Center` or `End`.

Reply:
I'll address the new behavior in the next PR.